Bottom: 3b88e934a65a9508e1abd6513ee82a9891c829f4
Top:    60a0f25bc239203d1ad53b9a06cffac284a89ad6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-06 15:04:54 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 181ba6e..fd68900 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -235,6 +235,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
